lib/std/channels.nim

```diff
@@ -445,12 +445,12 @@ proc `=destroy`*[T](c: var Channel[T]) =
 
 proc `=copy`*[T](dest: var Channel[T], src: Channel[T]) =
   ## Shares `Channel` by reference counting.
-  if src.d != nil:
-    atomicInc(src.d.atomicCounter)
-
-  if dest.d != nil:
-    `=destroy`(dest)
```

Comment:
I think the real bug is in calling =destroy here?

Reply:
not sure, what's wrong with it?

Reply:
Well, done right, this is how to do refcounting assignment:

```
  incRef source
  decRef dest
  dest = source
```

Since the incref comes first this is the natural protection against self assignments, no special casing required.

Reply:
yeah, and compare it to avoid unnecessary destruction and assignment.

Reply:
Self assignments are too rare to optimize for this case -- it's more efficient not to check at all here, slightly smaller code size, slightly better instruction cache usage.

Reply:
I see